### PR TITLE
API end point which returns list of groups for a given model

### PR DESCRIPTION
**Pull request by DravitLochan**
Fixes issue #246 
Fetches all the groups for a particular model. Right now the model is hard coded. 
Changes: added GroupListService.java file
screenshot from 2017-06-06 23-40-36

Screenshots for the change:

### CODE REVIEW

**Comment by DravitLochan** [comment]
@saurabhjn76 @dynamitechetan @Fluepke @jyoth1raditya Please review

**Comment by DravitLochan** [comment]
@saurabhjn76  [This](https://github.com/fossasia/susi_server/pull/257/commits/2e3841a8c7d55905a9612df8a169a6c7a8198c7f) commit message do contains a link to test it.

**Comment by DravitLochan** [comment]
@mariobehling Please review and merge.

**Comment by DravitLochan** [comment]
@mariobehling Please review and merge!

File: src/ai/susi/server/api/cms/GroupListService.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * Created by DravitLochan on 6/6/17.
+ * Servlet to load list of groups for given model
+ * Start the server and type in the below given address to test it.
+ *  http://127.0.0.1:4000/cms/getgroup.json
+ */
+public class GroupListService extends AbstractAPIHandler implements APIHandler{
+    @Override
+    public String getAPIPath() {
+        return "/cms/getgroup.json";
```
Comment (Orbiter) [inline src/ai/susi/server/api/cms/GroupListService.java]: lets call this "getgroups.json" (plural)

File: src/ai/susi/server/api/cms/GroupListService.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        return null;
+    }
+
+    @Override
+    public ServiceResponse serviceImpl(Query call, HttpServletResponse response, Authorization rights, JsonObjectWithDefault permissions) throws APIException {
+
+        String model_name = call.get("model", "general");
+        File model = new File(DAO.model_watch_dir, model_name);
+
+        String[] groups = model.list();
```
Comment (Orbiter) [inline src/ai/susi/server/api/cms/GroupListService.java]: please check each element in this list if it is actually a directory to filter out file name of files.
The list should only contain names of directories

**Comment by DravitLochan** [comment]
@Orbiter Please check. Updated the code as required.

**Comment by DravitLochan** [comment]
@mariobehling changes suggested are implemented. Please review and merge.

**Comment by DravitLochan** [comment]
@Orbiter Please see the changes you suggested are done. Kindly review and merge.

**Comment by DravitLochan** [comment]
This pr was increasing the number of commits unnecessarily so I made the changes and opened a new pr. 
Cheers

**Pull request closed without merging**